lib/netext/httpext/request.go

```diff
@@ -394,10 +432,17 @@ func MakeRequest(ctx context.Context, preq *ParsedHTTPRequest) (*Response, error
 
 	tracerTransport := newTransport(state, tags)
 	var transport http.RoundTripper = tracerTransport
-	if preq.Auth == "ntlm" {
-		transport = ntlmssp.Negotiator{
-			RoundTripper: tracerTransport,
-		}
+
+	// TODO: if HttpDebug is enabled, inject the debug tranport here? or use
+	// something like a virtual proxy for more accurate results, so we can catch
+	// things like HTTP/2 and exact headers? Connected issues:
+	// https://github.com/loadimpact/k6/issues/986,
+	// https://github.com/loadimpact/k6/issues/1042
```

Comment:
It will now also wrongly not report `digest` requests... Not that it reported them completely accurately before, but at least they were reported. But I agree, if we can't fix it quickly, we probably should add a warning for now... 

The solution I've proposed in this code would be enough to accurately report all requests (digest, ntlm, redirects, etc.), since it's going to happen at just one level above the underlying transport. Unfortunately, this won't be enough for https://github.com/loadimpact/k6/issues/986 and https://github.com/loadimpact/k6/issues/1042, since we'd probably still see HTTP1.1 and `Accept-Encoding`.

Though now that I'm thinking about this again, I'm wondering if this is a potential workaround even for that:
- implement the `HttpDebug` handling as a low-level `http.RoundTrripper` like it's mentioned above
- when a request is received, save its body and print only its method and URL and some sort of an ID, something like `[1234deadbeef] Making GET https://httpbin.org/anything ...`
- when we receive the response, we then use the `Response.Request` field to actually see what request was made? and print that?

Reply:
Unfortunately, that doesn't work, `Response.Proto` is updated to `HTTP/2.0`, but `response.Request.Proto` is `HTTP/1.1`. It seems like whatever we decide to do, we should ditch `DumpRequestOut` and roll our own... But the algorithm above, where we only print the request stub in the start and print the full request only after we receive the response still seems like the only way we can know if HTTP/2.0 was actually used...

Reply:
I'm wondering if I shouldn't do the simple `HttpDebug` dumping in this PR here, so we don't have a degradation, i.e. a roundtripper like this:
```go
dumpRequest(req)
res, err := originalTransport.RroundTrip(req)
dumpResponse(res)
return res, err
```
This won't solve #986 and #1042, but like the digest auth, at least it won't cause the PR to degrade what already existed?

Reply:
I did try to do this (waiting for the response) and than to print the request and found the same problem ... I am not certain if we should just print the request when we get the response instead of printing a stub , but let's move this discussion to a separate issue ?

Reply:
Sure, any of the already open issues would do... but what do you think about making my previous question - should I partially fix this here, so we don't have a degradation and at least print every HTTP request that is made?

Reply:
LGTM! 

apperantly I reloaded the pr last before you wrote it ;(

Reply:
Done